wallet-core/src/lib.rs

```diff
@@ -22,9 +22,35 @@ pub const RNG_SEED: usize = 64;
 // phoenix-transaction
 const MAX_INPUT_NOTES: usize = 4;
 
+use alloc::collections::btree_map::BTreeMap;
 use alloc::vec::Vec;
 
-use execution_core::transfer::phoenix::{Note, ViewKey as PhoenixViewKey};
+use execution_core::{
+    transfer::phoenix::{
+        Note, SecretKey as PhoenixSecretKey, ViewKey as PhoenixViewKey,
+    },
+    BlsScalar,
+};
+
+/// Filter all notes that are owned by the given keys, mapped to their
+/// nullifiers.
+pub fn filter_map(
```

Comment:
So, `filter_map` is actually what it does, but the name should be less generic. Worst case scenario, we should prepend a `notes_` or `phoenix_notes_` or something. But it would be nice if we actually find a better name.

Reply:
Since we have `notes` as a parameter already I would like to avoid having it in the name too. How about `filter_owned(notes, keys)`?

Reply:
Note: After a conversation we settled for the name `map_owned`.